### ISSUE TITLE
False positive when swapping elements of a matrix

### ISSUE THREAD

**Issue by krishna-veerareddy** [rust-lang/rust-clippy#4853]
Clippy(clippy 0.0.212 (3aea8603 2019-09-03)) incorrectly suggests to use `std::mem::swap` when trying to swap elements of a matrix across different rows. The lint is `clippy::manual_swap`.

Here's a minimal example:
```rust
fn swap(a: &mut[Vec<u64>]) {
    let tmp = a[0][1];

    a[0][1] = a[1][0];
    a[1][0] = tmp;
}
```

**Comment by flip1995** [comment]
It seems like #981 wasn't fixed completely.
